tests/test_node.py

```diff
@@ -20,3 +20,16 @@ def fn():
 
     with pytest.raises(ValueError):
         Node.from_fn(fn)
+
+
+def test_node_copy_with_retains_originating_functions():
+    def fn() -> int:
+        return 1
+
+    node = Node.from_fn(fn)
+    node_copy = node.copy_with(name="rename_fn", originating_functions=(fn,))
```

Comment:
You shouldn't need `originating_functions=(fn,)` here right?

Reply:
Nope its currently assigned in `resolve_nodes`